File: core/src/net/natruid/jungle/events/UnitHealthChangedEvent.java
package net.natruid.jungle.events;

import com.artemis.annotations.EntityId;

public class UnitHealthChangedEvent extends PooledEvent {
    @EntityId
    public int unit = -1;
    @EntityId
    public int source = -1;
    public int amount = 0;
    public boolean killed = false;

    @Override
    public void reset() {
        unit = -1;
        source = -1;
        amount = 0;
        killed = false;
    }
}

File: core/src/net/natruid/jungle/events/UnitHealthChangedEvent.kt
package net.natruid.jungle.events

class UnitHealthChangedEvent : PooledEvent() {
    var unit = -1
    var source = -1
    var amount = 0
    var killed = false

    override fun reset() {
        unit = -1
        source = -1
        amount = 0
        killed = false
    }
}
